gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/transformer/RequestPreProcessor.java

```diff
@@ -109,10 +110,23 @@ public EntitiesRequest transform(
         transformFilter(attributeIdMappings, originalRequest.getFilter(), context);
     Filter filter = mergeFilters(filterToInject, transformedFilter);
 
+    // Convert the time range into a filter and set it on the request so that all downstream
+    // components needn't treat it specially.
+    String timestampAttributeId = AttributeMetadataUtil.getTimestampAttributeId(
+        attributeMetadataProvider, context, originalRequest.getEntityType());
+
+    Filter.Builder filterBuilder = Filter.newBuilder()
+        .setOperator(AND)
+        .addChildFilter(getTimestampFilter(timestampAttributeId, Operator.GE, originalRequest.getStartTimeMillis()))
```

Comment:
Also if you do this, what about requests going to EDS? They will have time range in the filter as well and that might cause issues since those columns dont map to anything in entity attributes.

Reply:
Timestamp columns will only have QS as the source so they'd never go to EDS.

Reply:
But at this point you do not know where the query will go to. And if this filter hits the ExecutionTree then it ends up being split into two even if all the columns can be sourced from EDS. Because the time range is the filter now another query is made to QS just to get ids. And then the join happens and I'm not sure about this but you might return empty results since there's nothing from QS when there's no data?

Reply:
I think this is a separate but important question around liveness - even if all selections are time agnostic, shouldn't the returned entities be the same whether or not a metric is requested? In other words, I'd say this behavior was a bug, and we should never serve a response solely from ES.

Reply:
Yes, liveness needs to be addressed separately but @tim-mwangi what you're describing should be the current behavior. As long as there is time range in the query, we should return entities only if they had data in that time range. And, that will be maintained with this change.